src-cpp/rdkafkacpp.h

```diff
@@ -1335,6 +1345,34 @@ class RD_EXPORT KafkaConsumer : public virtual Handle {
    */
   virtual Message *consume (int timeout_ms) = 0;
 
+
+        /**
+         * @brief Consume message or get error event, triggers callbacks from
+         *        a queue.
+         *
+         * Will automatically call registered callbacks for any such queued
+         * events, including RdKafka::RebalanceCb, RdKafka::EventCb,
+         * RdKafka::OffsetCommitCb, etc.
+         *
+         * @remark Use \c delete to free the message.
+         *
+         * @remark  An application should make sure to call consume() at regular
+         *          intervals, even if no messages are expected, to serve any
+         *          queued callbacks waiting to be called. This is especially
+         *          important when a RebalanceCb has been registered as it needs
+         *          to be called and handled properly to synchronize internal
+         *          consumer state.
+         *
+         * @remark Application MUST NOT call \p poll() on KafkaConsumer objects.
+         *
+         * @returns One of:
+         *  - proper message (RdKafka::Message::err() is ERR_NO_ERROR)
+         *  - error event (RdKafka::Message::err() is != ERR_NO_ERROR)
+         *  - timeout due to no message or event in \p timeout_ms
+         *    (RdKafka::Message::err() is ERR__TIMED_OUT)
+         */
+        virtual Message *consume (Queue *queue, int timeout_ms) = 0;
```

Comment:
I'm not sure this belongs here since queues can be used by the legacy consumer too.
What about moving consume() to the Queue?